Snakefile

```diff
@@ -296,20 +401,30 @@ rule notify_genbank:
         shell("./bin/notify-on-location-hierarchy-addition {input.location_hierarchy} source-data/location_hierarchy.tsv")
         shell("./bin/notify-on-duplicate-biosample-change {input.duplicate_biosample} {params.s3_bucket}/duplicate_biosample.txt.gz")
 
-files_to_upload = {
-                    "metadata.tsv.gz":              f"data/{database}/metadata.tsv",
-                    "sequences.fasta.xz":           f"data/{database}/sequences.fasta",
-                    "nextclade.tsv.gz":             f"data/{database}/nextclade.tsv"}
-if database=="genbank":
-    files_to_upload["biosample.tsv.gz"] =           f"data/{database}/biosample.tsv"
-    files_to_upload["duplicate_biosample.txt.gz"] = f"data/{database}/duplicate_biosample.txt"
-elif database=="gisaid":
-    files_to_upload["additional_info.tsv.gz"] =     f"data/{database}/additional_info.tsv"
-    files_to_upload["flagged_metadata.txt.gz"] =    f"data/{database}/flagged_metadata.txt"
+
+def compute_files_to_upload(wildcards):
+    files_to_upload = {
+                        "metadata.tsv.gz":              f"data/{database}/metadata.tsv",
+                        "sequences.fasta.xz":           f"data/{database}/sequences.fasta"}
+    if database=="genbank":
+        files_to_upload["biosample.tsv.gz"] =           f"data/{database}/biosample.tsv"
+        files_to_upload["duplicate_biosample.txt.gz"] = f"data/{database}/duplicate_biosample.txt"
+    elif database=="gisaid":
+        files_to_upload["additional_info.tsv.gz"] =     f"data/{database}/additional_info.tsv"
+        files_to_upload["flagged_metadata.txt.gz"] =    f"data/{database}/flagged_metadata.txt"
+
+    nextclade_sequences_path = checkpoints.get_sequences_without_nextclade_annotations.get().output.fasta
+    if os.path.getsize(nextclade_sequences_path) > 0:
+        files_to_upload["nextclade.tsv.gz"] =                  f"data/{database}/nextclade.tsv"
+        files_to_upload["nextclade.mutation_summary.tsv.gz"] = f"data/{database}/nextclade.mutation_summary.tsv"
+        files_to_upload["nextclade.aligned.fasta.xz"] =        f"data/{database}/nextclade.aligned.fasta"
+
+    return files_to_upload
```

Comment:
I thing that the latest failure was due to missing return statement here. The `unpack()` on that return would then operate on a `None`. Let's see if it fixes things.

Reply:
Alright, GenBank ran successfully the "now work" path if I understand correctly. Although I could not find confirmation for that in the logs. There is no mention of Nextclade though. `genbank.ndjson.xz` was identical, but `biosample.ndjson.gz` was not. I am not sure how biosample data is used. Is this correct behavior?

GISAID run failed with 5 download attempts, so inconclusive.

Let's try again? I merged master into this branch this time, to take advantage of Moira's cleanup in annotations, so the logs should not consist mostly of the "REDUNDANT ANNOTATED METADATA" messages.
 
https://github.com/nextstrain/ncov-ingest/actions/runs/1583829615
https://github.com/nextstrain/ncov-ingest/actions/runs/1583829620

Reply:
In the successful GenBank run, I see the pipeline jumped directly from `checkpoint get_sequences_without_nextclade_annotations` to `rule generate_metadata`. If the `genbank.ndjson.xz` was identical, that means there were not any new sequences. The `biosample.ndjson.gz` only adds additional metadata, it does not contain any sequences. This worked as expected (at least for this path) 🎉

Reply:
Thanks for catching my missing `return`! 

Looks like we've seen it correctly skip steps (when there are no new sequences) so the next trial run should test out the other code path 🤞